cachelib/adaptor/rocks_secondary_cache/CachelibWrapper.cpp

```diff
@@ -225,64 +286,31 @@ void RocksCachelibWrapper::Close() {
     // sections already started to finish, and then delete the cache
     cache_.store(nullptr);
     GetRcuDomain().synchronize();
-    admin_.reset();
     delete cache;
   }
 }
 
-bool RocksCachelibWrapper::UpdateMaxWriteRateForDynamicRandom(
-    uint64_t maxRate) {
-  FbCache* cache = cache_.load();
-  bool ret = false;
-  if (cache) {
-    ret = ApiWrapper::updateMaxRateForDynamicRandomAP(*cache, maxRate);
-  }
-  return ret;
-}
-
 // Global cache object and a default cache pool
 std::unique_ptr<rocksdb::SecondaryCache> NewRocksCachelibWrapper(
     const RocksCachelibOptions& opts) {
-  std::unique_ptr<FbCache> cache;
-  std::unique_ptr<cachelib::CacheAdmin> admin;
-  cachelib::PoolId defaultPool;
-  FbCacheConfig config;
-  NvmCacheConfig nvmConfig;
-
-  nvmConfig.navyConfig.setBlockSize(opts.blockSize);
-  nvmConfig.navyConfig.setSimpleFile(opts.fileName,
-                                     opts.size,
-                                     /*truncateFile=*/true);
-  nvmConfig.navyConfig.blockCache().setRegionSize(opts.regionSize);
-  if (opts.admPolicy == "random") {
-    nvmConfig.navyConfig.enableRandomAdmPolicy().setAdmProbability(
-        opts.admProbability);
-  } else {
-    nvmConfig.navyConfig.enableDynamicRandomAdmPolicy()
-        .setMaxWriteRate(opts.maxWriteRate)
-        .setAdmWriteRate(opts.admissionWriteRate);
-  }
-  nvmConfig.enableFastNegativeLookups = true;
-
-  config.setCacheSize(opts.volatileSize)
-      .setCacheName(opts.cacheName)
-      .setAccessConfig(
-          {opts.bktPower /* bucket power */, opts.lockPower /* lock power */})
-      .enableNvmCache(nvmConfig)
-      .validate(); // will throw if bad config
-  cache = std::make_unique<FbCache>(config);
-  defaultPool =
-      cache->addPool("default", cache->getCacheMemoryStats().cacheSize);
-
-  if (opts.fb303Stats) {
```

Comment:
Is CacheAdmin for stats not supported?

Reply:
@anand1976 This is another piece that is internal to Facebook/Meta and is not in the public API.